### PR TITLE
Assure a default toolset is always set.

### PR DESCRIPTION
**Pull request by tvandijck**
We're running into the issue where:
```
filter { 'toolset:msc-v140' }
```
doesn't work, because the majority of time, the toolset is not actually set. This change assures the filter always has a valid value in it, and not just 'nil' in the majority of cases.

We also normalize the toolset, so when you do
```
toolset 'v140'
```
it normalizes to ```msc-v140```, as well as the other rules implemented in the ```p.tool.canonical``` function.
We should realy deprecate the v140 naming however, and require explicit msc-v140 usage, so I added a p.warnOnce in there too.

### CODE REVIEW

File: src/base/tools.lua [hunk 1]
```diff
@@ -29,25 +29,34 @@
...
-				if parts[1] == "msc" and tonumber(parts[2]:sub(1,3)) ~= nil then
-					parts[2] = "v" .. parts[2]
-				end
-
-				-- perform case-correction of the LLVM toolset
-				if parts[2]:startswith("llvm-vs") then
-					parts[2] = "LLVM-" .. parts[2]:sub(6)
-				end
-			end
+			p.warnOnce(identifier, "toolset '%s' is deprecated, toolset 'msc-%s' instead.", identifier, identifier)
```
Comment (starkos) [inline src/base/tools.lua]: That was the original intention behind leaving the "v140" form in place: that matches what VS uses, and what you see in the IDE, so it is a more "expected" format if you're familiar with the VS projects. We did go back and forth on it though.

Comment (tvandijck) [inline src/base/tools.lua]: I just put the deprecation message in there, because there is a comment that says "we should deprecate this"...

**Comment by tvandijck** [comment]
Removed the deprecation warning.

**Comment by tvandijck** [comment]
Removed the deprecation warning a while ago... need more feedback however on what direction we want to go for the normalization?

**Pull request merged**